repos/fdbt-site/src/pages/capConfirmation.tsx

```diff
@@ -0,0 +1,229 @@
+        });
+    }
+
+    if (tapsPricingContents.length > 0) {
+        confirmationElements.push({
+            name: 'Prices by Taps',
+            content: tapsPricingContents,
+            href: '/multiTapsPricing',
+        });
+    }
+
+    if (capDistancePricingContents.length > 0) {
+        confirmationElements.push({
+            name: 'Prices',
+            content: capDistancePricingContents,
+            href: '/defineCapPricingPerDistance',
+        });
+    }
+
+    return confirmationElements;
+};
+
+const CapConfirmation = ({
+    typeOfCap,
+    productGroupName,
+    caps,
+    capValidity,
+    capStartInfo,
+    services,
+    tapsPricingContents,
+    capDistancePricingContents,
+    csrfToken,
+}: CapConfirmationProps): ReactElement => (
+    <TwoThirdsLayout title={title} description={description} errors={[]}>
+        <CsrfForm action="/api/capConfirmation" method="post" csrfToken={csrfToken}>
+            <>
+                <h1 className="govuk-heading-l">Check your answers before sending your fares information</h1>
+                <ConfirmationTable
+                    header="Cap Information"
+                    confirmationElements={buildCapConfirmationElements(
+                        typeOfCap,
+                        productGroupName,
+                        caps,
+                        capValidity,
+                        capStartInfo,
+                        services,
+                        tapsPricingContents,
+                        capDistancePricingContents,
+                    )}
+                />
+                <input type="submit" value="Continue" id="continue-button" className="govuk-button" />
+            </>
+        </CsrfForm>
+    </TwoThirdsLayout>
+);
+
+export const getServerSideProps = async (ctx: NextPageContextWithSession): Promise<{ props: CapConfirmationProps }> => {
+    const csrfToken = getCsrfToken(ctx);
+    const noc = getAndValidateNoc(ctx);
+
+    const typeOfCapAttribute = getSessionAttribute(ctx.req, TYPE_OF_CAP_ATTRIBUTE);
+    const capAttribute = getSessionAttribute(ctx.req, CAPS_ATTRIBUTE);
+    const productGroupIdAttribute = getSessionAttribute(ctx.req, CAPPED_PRODUCT_GROUP_ID_ATTRIBUTE);
+
+    const productGroupName =
+        productGroupIdAttribute && typeof productGroupIdAttribute === 'string'
+            ? (await getProductGroupByNocAndId(noc, Number.parseInt(productGroupIdAttribute)))?.name
+            : '';
+    const caps = capAttribute ? capAttribute.caps : [];
+
+    const capValidityAttribute = getSessionAttribute(ctx.req, CAP_EXPIRY_ATTRIBUTE);
+    const capValidity =
+        capValidityAttribute && isCapExpiry(capValidityAttribute)
+            ? `${capValidityAttribute.productValidity}${
+                  capValidityAttribute.productEndTime ? ` - ${capValidityAttribute.productEndTime}` : ''
+              }`
+            : '';
+
+    const capStartAttribute = getSessionAttribute(ctx.req, CAP_START_ATTRIBUTE);
+
+    if (
```

Comment:
This check needs to happen higher up in the function. We should get all the attributes we know we need, and then do this check to make sure they're all present.

Reply:
@NitishAroraKPMG  are we not going to do this?